game/src/main/kotlin/jake2/game/adapters/SuperAdapter.kt

```diff
@@ -0,0 +1,180 @@
+/*
+Copyright (C) 1997-2001 Id Software, Inc.
+
+This program is free software; you can redistribute it and/or
+modify it under the terms of the GNU General Public License
+as published by the Free Software Foundation; either version 2
+of the License, or (at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  
+
+See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program; if not, write to the Free Software
+Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
+
+*/
+// Created on 09.01.2004 by RST.
+package jake2.game.adapters
+
+import jake2.game.GameExportsImpl
+import jake2.game.SubgameEntity
+import jake2.qcommon.cplane_t
+import jake2.qcommon.csurface_t
+import jake2.qcommon.filesystem.QuakeFile
+
+/**
+ * There are many fields in the [SubgameEntity] class. Some of them represent state - like 'health', 'damage' etc,
+ * other represent behaviour - like 'think', 'touch', 'use' etc.
+ * In c version these were function pointers.
+ *
+ *
+ * The purpose of all Adapter registration is to store and restore such behavioural edict fields.
+ */
+abstract class SuperAdapter {
+    /** Constructor, does the adapter registration.  */
```

Comment:
I'd remove obvious comments from this class :)

Reply:
fixed